ext/MitsubaLoader/CMitsubaLoader.h

```diff
@@ -16,6 +16,304 @@ namespace ext
+	_IRR_STATIC_INLINE_CONSTEXPR uint32_t BITFIELDS_MASK_SIGMA_A_TEX = 0x1u;
+	_IRR_STATIC_INLINE_CONSTEXPR uint32_t BITFIELDS_SHIFT_SIGMA_A_TEX = 15u;
+	_IRR_STATIC_INLINE_CONSTEXPR uint32_t BITFIELDS_MASK_WEIGHT_TEX = 0x1u;
+	_IRR_STATIC_INLINE_CONSTEXPR uint32_t BITFIELDS_SHIFT_WEIGHT_TEX = 10u;
+	_IRR_STATIC_INLINE_CONSTEXPR uint32_t BITFIELDS_MASK_OPACITY_TEX = 0x1u;
+	_IRR_STATIC_INLINE_CONSTEXPR uint32_t BITFIELDS_SHIFT_OPACITY_TEX = 10u;
+	_IRR_STATIC_INLINE_CONSTEXPR uint32_t BITFIELDS_MASK_PHONG_EXP_TEX = 0x1u;
+	_IRR_STATIC_INLINE_CONSTEXPR uint32_t BITFIELDS_SHIFT_PHONG_EXP_TEX = 16u;
+	_IRR_STATIC_INLINE_CONSTEXPR uint32_t BITFIELDS_MASK_MIXTURE_CHILD_COUNT = 0x1fu;
+	_IRR_STATIC_INLINE_CONSTEXPR uint32_t BITFIELDS_SHIFT_MIXTURE_CHILD_COUNT = 10u;
+#include "irr/irrpack.h"
+	/**
+	GLSL:
+	struct {
+		uint bitfields;
+		vec2 alpha;
+		vec3 reflectance;
+	};
+	*/
+	struct alignas(16) SAllDiffuse
+	{
+		uint32_t bitfields;
+		float dummy;
+		//if flag decides to use alpha texture, {alpha[0..1]} is bindless texture ID (bit-cast to uvec2)
+		//otherwise alpha[0] is single-float alpha
+		float alpha[2];
+		//if flag decides to use reflectance texture, {reflectance[0..1]} is bindless texture ID (bit-cast to uvec2)
+		//otherwise reflectance[0..2] is constant reflectance
+		float reflectance[3];
+	} PACK_STRUCT;
+	/**
+	GLSL:
+	struct {
+		uint bitfields;
+		vec3 transmittance;
+	};
+	*/
+	struct alignas(16) SDiffuseTransmitter
+	{
+		uint32_t bitfields;
+		float dummy[3];
+		//if flag decides to use transmittance texture, {transmittance[0..1]} is bindless texture ID (bit-cast to uvec2)
+		//otherwise transmittance[0..2] is constant transmittance
+		float transmittance[3];
+	} PACK_STRUCT;
+	/**
+	GLSL:
+	struct {
+		uint bitfields;
+		vec2 int_ext_ior;
+		vec2 alpha_u;
+		vec2 alpha_v;
+		vec3 specReflectance;
+		vec3 specTransmittance;
+	}
+	*/
+	struct alignas(16) SAllDielectric
+	{
+		uint32_t bitfields;
+		float dummy0;
+		//int_ext_ior[0] is internal IoR, int_ext_ior[1] is external IoR
+		float int_ext_ior[2];
+		//if NDF is Ashikhmin-Shirley:
+		//	if flag decides to use alpha_u texture, alpha_u[0..1] is bindless texture ID
+		//	otherwise alpha_u[0] is constant alpha_u
+		//	if flag decides to use alpha_v texture, alpha_v[0..1] is bindless texture ID
+		//	otherwise alpha_v[0] is constant alpha_v
+		//otherwise (different NDF)
+		//	if flag decides to use alpha texture, alpha_u[0..1] is bindless texture ID
+		//	otherwise alpha_u[0] is constant alpha
+		float alpha_u[2];
+		float alpha_v[2];
+		//if flag decides to use spec refl texture, specReflectance[0..1] is bindless texture ID
+		//otherwise specReflectance[0..2] is constant spec refl
+		float specReflectance[3];
+		float dummy1;
+		//if flag decides to use spec transmitance texture, specTransmittance[0..1] is bindless texture ID
+		//otherwise specTransmittance[0..2] is constant spec transmitance
+		float specTransmittance[3];
```

Comment:
I know that Mitsuba considers a non-PBR override (constant reflectance and transmittance) of the Dielectric Material the same BSDF, but we should dump that into a special case separate from an actual PBR Dielectric

Reply:
we could actually error out if reflectance or transmittance is being used... no reason to support non PBR BSDFs.